Steamfitter.Api/Services/TaskExecutionService.cs

```diff
@@ -537,6 +549,66 @@ private STT.Task<string> RunTask(TaskEntity taskToExecute, ResultEntity resultEn
             return task;
         }
 
+        private async STT.Task<string> HttpTaskTask(TaskEntity taskToExecute)
+        {
+            HttpResponseMessage response;
+            using (var scope = _scopeFactory.CreateScope())
+            {
+                var tokenResponse = await ApiClientsExtensions.GetToken(scope);
+                var actionParameters = JsonSerializer.Deserialize<HttpInputString>(taskToExecute.InputString);
+                var url = actionParameters.Url;
+                if (url.Contains("?"))
+                {
+                    url += $"&asuser={taskToExecute.UserId.ToString()}" ;
+                }
+                else
+                {
+                    url += $"?asuser={taskToExecute.UserId.ToString()}" ;
+                }
+                var client = ApiClientsExtensions.GetHttpClient(_httpClientFactory, url, tokenResponse);
+                switch (taskToExecute.Action)
+                {
+                    case TaskAction.http_get:
+                        {
+                            response = await client.GetAsync(url);
+                            break;
+                        }
+                    case TaskAction.http_post:
+                        {
+                            StringContent content = new StringContent(actionParameters.Body, Encoding.UTF8, "application/json");
+                            client.DefaultRequestHeaders.Add("Accept", "application/json");
+                            response = await client.PostAsync(taskToExecute.VmMask, content);
+                            break;
+                        }
+                    case TaskAction.http_put:
+                        {
+                            StringContent content = new StringContent(actionParameters.Body);
+                            response = await client.PutAsync(taskToExecute.VmMask, content);
+                            break;
+                        }
+                    case TaskAction.http_delete:
+                        {
+                            response = await client.DeleteAsync("http://www.google.com");
+                            break;
+                        }
+                    default:
+                        {
+                            return $"Action '{taskToExecute.Action.ToString()}' is not a valid action for http tasks";
+                        }
+                }
+                if (response.IsSuccessStatusCode)
+                {
+                    var responseString = await response.Content.ReadAsStringAsync();
+                    return responseString;
```

Comment:
might be useful to return the response code itself as well. I could see a Success condition being based on the response code rather than the content for these types of tasks. Probably a separate feature though.

Reply:
added a TODO